packages/core/src/singleton/API/types.ts

```diff
@@ -7,8 +7,9 @@ export type LibraryAPIOptions = {
 		// custom headers for given GraphQL service. Will be applied to all operations.
 		headers?: (options: {
 			query: string;
-			variables: Record<string, DocumentType>;
+			variables?: Record<string, DocumentType>;
 		}) => Promise<Headers>;
+		withCredentials: boolean;
```

Comment:
Thanks!! 

I think this should be optional:
```suggestion
		withCredentials?: boolean;
```

Reply:
Good catch!!